src/scales/scale.timeseries.js

```diff
@@ -72,18 +64,53 @@ class TimeSeriesScale extends TimeScale {
 
 		items.push(max);
 
-		for (i = 0, ilen = items.length; i < ilen; ++i) {
-			next = items[i + 1];
-			prev = items[i - 1];
-			curr = items[i];
+		return items;
+	}
 
-			// only add points that breaks the scale linearity
-			if (prev === undefined || next === undefined || Math.round((next + prev) / 2) !== curr) {
-				table.push({time: curr, pos: i / (ilen - 1)});
-			}
+	/**
+	 * Returns all timestamps
+	 * @private
+	 */
+	_getTimestampsForTable() {
+		const me = this;
+		let timestamps = me._cache.all || [];
+
+		if (timestamps.length) {
+			return timestamps;
+		}
+
+		const data = me.getDataTimestamps();
+		const label = me.getLabelTimestamps();
+		if (data.length && label.length) {
+			// If combining labels and data (data might not contain all labels),
+			// we need to recheck uniqueness and sort
+			timestamps = _arrayUnique(data.concat(label).sort(sorter));
+		} else {
+			timestamps = data.length ? data : label;
 		}
+		timestamps = me._cache.all = timestamps;
 
-		return table;
+		return timestamps;
+	}
+
+	/**
+	 * @param {number} value - Milliseconds since epoch (1 January 1970 00:00:00 UTC)
+	 * @return {number}
+	 */
+	getDecimalForValue(value) {
+		const table = this._table;
+		const {lo, hi} = _lookup(table, value);
+
+		// Note: the lookup table ALWAYS contains at least 2 items (min and max)
+		const prev = table[lo];
+		const next = table[hi];
+
+		const span = next - prev;
+		const ratio = span ? (value - prev) / span : 0;
+		const maxIndex = (table.length - 1);
+		const offset = (hi / maxIndex - lo / maxIndex) * ratio;
```

Comment:
I think the logic is the same. It's just more obvious now how complicated it was in the first place because I inlined `interpolate`. I think this method should be simplified as well by only supporting values that are in the table similar to `getValueForPixel` and the category scale